README.rst

```diff
@@ -29,57 +31,89 @@ Further documentation is available on
+        CurationRequestSubmitNotificationBuilder.type: CurationRequestSubmitNotificationBuilder,
+    }
 
-## Overwrite deposit view
-The deposit view has to be updated to include the curation section. Most importantly, the curation specific javascript has to be included in the javascript block:
-`{{ webpack['invenio-curations-deposit.js'] }}`
+    # enable requests to target groups of users (i.e. `roles`)
+    NOTIFICATIONS_ENTITY_RESOLVERS = NOTIFICATIONS_ENTITY_RESOLVERS + [
+        ServiceResultResolver(service_id="groups", type_key="group")
+    ]
 
-This can be achieved by providing a custom template. Make sure to copy the current template from `invenio_app_rdm/records_ui/templates/semantic-ui/invenio_app_rdm/records/deposit.html`.
-In order to override this, place it into your instances `templates/semantic-ui/invenio_app_rdm/records/deposit.html` folder.
-Then add the afore mentioned line to the javascript block in the template
-```
-{%- block javascript %}
-  {{ super() }}
-  ...
 
-  {# This line right here #}
-  {{ webpack['invenio-curations-deposit.js'] }}
-{%- endblock %}
+Add service component
+^^^^^^^^^^^^^^^^^^^^^
 
-```
+In order to require an accepted curation request before publishing a record, the component has to be appended to the RDM record service:
 
-## Create curator role
-A curator role is used to ensure a user has the permission to manage curation requests. Its name is specified via a config variable `CURATIONS_MODERATION_ROLE`.
-In order to create this role, the following command can be run inside an instance: `invenio roles create <name-of-curation-role>`
-Adding a role to a user can be achieved by running: `invenio roles add <user-email-address> <name-of-curation-role>`
+.. code-block:: python
+
+    from invenio_curations.services.components import CurationComponent
+    from invenio_rdm_records.services.components import DefaultRecordsComponents
+
+    # NOTE: the curation component should be added at the end
+    RDM_RECORDS_SERVICE_COMPONENTS = DefaultRecordsComponents + [
+        CurationComponent,
+    ]
+
+
+Overwrite deposit view template
+-------------------------------
+
+The deposit view has to be updated to include the curation section.
+Most importantly, the curation specific JavaScript has to be included in the JavaScript block:
+``{{ webpack['invenio-curations-deposit.js'] }}``
+
+This can be achieved by providing a custom template, e.g. in your instance's ``templates/`` directory:
+
+Copy the current template from ``invenio_app_rdm/records_ui/templates/semantic-ui/invenio_app_rdm/records/deposit.html`` (available e.g. `here <https://github.com/inveniosoftware/invenio-app-rdm/blob/master/invenio_app_rdm/records_ui/templates/semantic-ui/invenio_app_rdm/records/deposit.html>`_) into your instance's ``templates/`` directory (the last parts of the path have to match): ``templates/semantic-ui/invenio_app_rdm/records/deposit.html``.
```

Comment:
for the record, one of the reST quirks:

originally i tried to create a link with code formatting:
~~~rest
```invenio_app_rdm/records_ui/templates/semantic-ui/invenio_app_rdm/records/deposit.html`` <https://github.com/inveniosoftware/invenio-app-rdm/blob/master/invenio_app_rdm/records_ui/templates/semantic-ui/invenio_app_rdm/records/deposit.html>`_
~~~
but it looks like sphinx couldn't handle the triple back-ticks in the beginning and rendered weird stuff.

in markdown, this is simple:
[`invenio_app_rdm/records_ui/templates/semantic-ui/invenio_app_rdm/records/deposit.html`](https://github.com/inveniosoftware/invenio-app-rdm/blob/master/invenio_app_rdm/records_ui/templates/semantic-ui/invenio_app_rdm/records/deposit.html)

```md
[`invenio_app_rdm/records_ui/templates/semantic-ui/invenio_app_rdm/records/deposit.html`](https://github.com/inveniosoftware/invenio-app-rdm/blob/master/invenio_app_rdm/records_ui/templates/semantic-ui/invenio_app_rdm/records/deposit.html)
```